charmcraft/models/project.py

```diff
@@ -98,6 +102,19 @@ def from_bases(cls: type[Self], bases: Iterable[charmcraft.Base]) -> Self:
         return cls("_".join(base_strings))
 
 
+class Platform(models.CraftBaseModel):
+    """Project platform definition."""
+
+    build_on: list[CharmArch] = pydantic.Field(min_items=1)
+    build_for: list[CharmArch | Literal["all"]] = pydantic.Field(min_items=1, max_items=1)
```

Comment:
Did charmcraft previously support `all`?

Reply:
If you didn't provide an architecture in a `build-for` base it would declare the charm to run on all architectures (which is problematic because that meant the implicit default did that, but if you brought in rust deps it wouldn't actually run on all architectures). So now we're allowing your to specify your charm as arch-independent, but it's not doing it by default.